x-pack/legacy/plugins/monitoring/public/views/loading/index.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { render, unmountComponentAtNode } from 'react-dom';
+import { PageLoading } from 'plugins/monitoring/components';
+import uiRoutes from 'ui/routes';
+import { I18nContext } from 'ui/i18n';
+import template from './index.html';
+import { setAngularState, toggleSetupMode, getSetupModeState, initSetupModeState } from '../../lib/setup_mode';
+
+const REACT_DOM_ID = 'monitoringLoadingReactApp';
+
+uiRoutes
+  .when('/loading', {
+    template,
+    controller: class {
+      constructor($injector, $scope) {
+        const monitoringClusters = $injector.get('monitoringClusters');
+        const kbnUrl = $injector.get('kbnUrl');
+
+        setAngularState($scope, $injector);
+        initSetupModeState();
+
+        const setupMode = getSetupModeState();
+        // For phase 3, this is not an valid route unless
+        // setup mode is currently enabled. For phase 4,
+        // we will remove this check.
+        if (!setupMode.enabled) {
+          kbnUrl.changePath('/no-data');
+          return;
+        }
+
+        $scope.$on('$destroy', () => {
+          unmountComponentAtNode(document.getElementById(REACT_DOM_ID));
+        });
+
+        $scope.$$postDigest(() => {
+          this.renderReact();
+        });
+
+        monitoringClusters()
+          .then(clusters => {
+            if (clusters && clusters.length) {
+              kbnUrl.changePath('/home');
+              return;
+            }
+            setAngularState($scope, $injector);
+            return toggleSetupMode(true)
+              .then(() => {
+                kbnUrl.changePath('/elasticsearch/nodes');
+                $scope.$apply();
```

Comment:
I think `kbnUrl.changePath()` will already do a digest, so probably no need for `$scope.$apply()`

Reply:
If that's the case, would I see an error about already in the digest loop?

Reply:
It might be done one after the other, but it does not really matter since `kbnUrl.changePath` will completely tear down that component/page and replace it with the new components. This means that everything will already be updated and synced